stdlib/SparseArrays/test/sparse.jl

```diff
@@ -339,6 +339,12 @@ end
     end
 end
 
+@testset "Issue #30502" begin
+    @test sprand(UInt8(16), UInt8(16), 1.0) != spzeros(UInt8(16), UInt8(16))
+    @test sprand(Int8(16), Int8(16), 1.0) != spzeros(Int8(16), Int8(16))
+    @test sprand(UInt8(16), UInt8(16), 1.0, ones) != spzeros(UInt8(16), UInt8(16))
+end
```

Comment:
Can we have a test that checks correctness of the intended answer with `==`? Also, only one test would be enough here.